src/execution/operator/scan/physical_table_scan.cpp

```diff
@@ -27,14 +48,35 @@ void PhysicalTableScan::GetChunkInternal(ClientContext &context, DataChunk &chun
 		table.InitializeScan(transaction, state->scan_offset, column_ids);
 		state->initialized = true;
 	}
-
-	table.Scan(transaction, chunk, state->scan_offset);
+	//! Get max value
+    idx_t result_count = -1;
+    do{
+        chunk.Reset();
+        table.Scan(transaction, chunk, state->scan_offset, table_filters);
+        if (expression){
+            SelectionVector sel(STANDARD_VECTOR_SIZE);
+            idx_t initial_count = chunk.size();
+            result_count = state->executor.SelectExpression(chunk, sel);
+            if (result_count == initial_count) {
+                //! Nothing was filtered: skip adding any selection vectors
+                return;
+            }
+            chunk.Slice(sel, result_count);
+        }
+    } while (result_count == 0 && state->scan_offset.current_transient_row < state->scan_offset.max_transient_row);
```

Comment:
The condition here seems weird: why are you checking the state->scan_offset.current_transient_row? It's better not to peek into the scan state here, the DataTable::Scan already performs this check. The check should just be removed.